client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -321,6 +333,31 @@ class DeltaSharingRestClient(
     )
   }
 
+  private def getFilesByPage(
+      targetUrl: String,
+      request: QueryTableRequest): (Long, String, Seq[String]) = {
+    val allLines = ArrayBuffer[String]()
+    var updatedRequest = request.copy(maxFiles = Some(maxFilesPerReq))
+    var (version, respondedFormat, lines) = getNDJson(targetUrl, updatedRequest)
+    allLines.appendAll(lines)
+    var nextPageToken = JsonUtils.fromJson[SingleAction](lines.last).nextPageToken
+    while (nextPageToken != null && nextPageToken.token != null && nextPageToken.token.nonEmpty) {
+      // remove last line (nextPageToken) in previous page
+      allLines.remove(allLines.length - 1)
+      updatedRequest = updatedRequest.copy(pageToken = Some(nextPageToken.token))
+      lines = getNDJson(targetUrl, updatedRequest)._3
```

Comment:
Just wanna double check, it's safe to assume that the `respondedFormat` will be the same for multiple page calls right? Because we're using the same version of client and the header should have the same value for the `capabilities` tag.

Reply:
This also assumes the server behavior doesn't change across the time of the multiple calls, which will be true for most of the time.
it's good to add a check and throw error though.